code/modules/economy/account.dm

```diff
@@ -42,14 +43,25 @@
 /datum/bank_account/proc/payday(amt_of_paychecks, free = FALSE)
 	if(free)
 		adjust_money(account_job.paycheck * amt_of_paychecks)
+		if(welfare)
+			adjust_money(PAYCHECK_WELFARE * amt_of_paychecks)
 	else
 		var/datum/bank_account/D = SSeconomy.get_dep_account(account_job.paycheck_department)
 		if(D)
-			if(!transfer_money(D, account_job.paycheck * amt_of_paychecks))
-				bank_card_talk("ERROR: Payday aborted, departmental funds insufficient.")
-				return FALSE
+			if(!welfare)
```

Comment:
NO, do a total adjustment calculation ONCE and apply it everywhere

what the fuck is this garbage, you know better